src/pkg_exporter/textfile.py

```diff
@@ -50,12 +59,91 @@ def main():
     rebootmanager.query()
     reboot_gauge.set(rebootmanager.getMetricValue())
 
-    exporter_file = os.getenv("PKG_EXPORTER_FILE", "/var/prometheus/pkg-exporter.prom")
+
+def write_registry_to_file(registry, exporter_file=None):
+    if not exporter_file:
+        exporter_file = os.getenv(
+            "PKG_EXPORTER_FILE", "/var/prometheus/pkg-exporter.prom"
+        )
     exporter_dir = os.path.dirname(exporter_file)
     os.makedirs(exporter_dir, exist_ok=True)
 
     write_to_textfile(exporter_file, registry)
 
 
+def serve(addr, port, timewait, rootdir):
+    start_http_server(addr=addr, port=port)
+    while True:
+        sleep(timewait)
```

Comment:
Is there a more elegant way of solving this? I'm not a fan of a static sleep, especially since this does not include the time "populate_registry" takes.